RHDEVDOCS-6342: Content creation for sensitive annotations feature (#87882)
openshift/openshift-docs: securing_openshift_gitops/masking-sensitive-annotations-in-the-argo-cd-web-ui.adoc

:_mod-docs-content-type: ASSEMBLY
include::_attributes/common-attributes.adoc[]
[id="masking-sensitive-annotations-in-the-argo-cd-web-ui"]
= Masking sensitive annotations in the Argo CD Web UI
:context: masking-sensitive-annotations-in-the-argocd-web-ui

toc::[]

> Argo CD hides sensitive annotation values on Secret resources from the Web UI and CLI. Users configure this by masking specific annotation keys in the Argo CD Custom Resource (CR). This feature enhances security by preventing accidental exposure of sensitive information, such as tokens or API keys, stored in annotations on Secret resources.

Reviewer: ```suggestion
Argo CD hides sensitive annotation values on Secret resources from the Web UI and CLI. Users can configure this by specifying annotation keys to be masked in the Argo CD Custom Resource (CR). This feature enhances security by preventing accidental exposure of sensitive information, such as tokens or API keys, stored in annotations on Secret resources.
```

Author: Fixed.

> Argo CD hides sensitive annotation values on Secret resources from the Argo CD Web UI and CLI. Users can configure this by specifying annotation keys to be masked in the Argo CD Custom Resource (CR). This feature enhances security by preventing accidental exposure of sensitive information, such as tokens or API keys, stored in annotations on Secret resources.

Reviewer 2: ```suggestion
Argo CD hides sensitive annotation values on Secret resources from the Argo CD Web UI and CLI. Users can configure this by specifying annotation keys to be masked in the Argo CD custom resource (CR). This feature enhances security by preventing accidental exposure of sensitive information, such as tokens or API keys, stored in annotations on `Secret` resources.
```

IMO, "Secret" should be in back ticks. Alternatively you can use lower case.

Author: Fixed.

> Argo CD hides sensitive annotation values on `Secret` resources from the Argo CD Web UI and CLI. Users can configure this by specifying annotation keys to be masked in the Argo CD custom resource (CR). This feature enhances security by preventing accidental exposure of sensitive information, such as tokens or API keys, stored in annotations on `Secret` resources.

Author: Fixed.

To enable this feature, add the `resource.sensitive.mask.annotations` key under `.spec.extraConfig` in the Argo CD CR. Specify a comma-separated list of annotation keys to mask.

> [NOTE]

Author: Fixed.

====
Ensure that the annotation keys listed in `resource.sensitive.mask.annotations` are accurate and relevant to your use case. This feature does not support wildcards and requires explicit configuration in the Argo CD CR.
====

.Prerequisites

> * Ensure that you have created an Argo CD instance. For more information, see Installing a user-defined Argo CD instance.

Author: Fixed.

> * Ensure that you have created an Argo CD instance. For more information, see _Installing a user-defined Argo CD instance_.

Reviewer 2: ```suggestion
* Ensure that you have created an Argo CD instance. For more information, see "Installing a user-defined Argo CD instance".
```

https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#lead-in-link-sentence-style-for-non-xref-situations

Author: Fixed.

> * Ensure that you have created an Argo CD instance. For more information, see "Installing a user-defined Argo CD instance".

Reviewer 3: https://redhat-documentation.github.io/supplementary-style-guide/#prerequisites

> * Ensure that you have created an Argo CD instance. For more information, see "Installing a user-defined Argo CD instance".

Author: Fixed.

// Enabling sensitive annotations masking in the Argo CD Web UI
include::modules/gitops_enabling_sensitive_annotations_in_the_argo_cd_web_ui.adoc[leveloffset=+1]

[role="_additional-resources"]
[id="additional-resources_{context}"]
== Additional resources

> * xref:../argocd_instance/setting-up-argocd-instance.adoc#gitops-argo-cd-installation[Installing a user-defined Argo CD instance]

Reviewer 4: Should we mention upstream documentation?
https://argo-cd.readthedocs.io/en/stable/operator-manual/declarative-setup/#mask-sensitive-annotations-on-secrets
